quiltctl/command/init/initializer.js

```
@@ -0,0 +1,213 @@
+  *
+  * @return {boolean} True on success, false on failure.
+  */
+function sshGenerateKeyPair() {
+  try {
+    execSync(
+      `ssh-keygen -t rsa -b 2048 -f ${consts.quiltSshKeyLocationPrivate}` +
+      ` -N '' -C ''`,
+      {stdio: 'ignore'});
+    return true;
+  } catch (err) {
+    return false;
+  }
+}
+
+/**
+  * Create new credentials file if needed.
+  *
+  * @param {Provider} provider The chosen provider.
+  * @param {Promise} answers A promise that contains the user's answers.
+  * @return {void} Returns nothing.
+  */
+function writeProviderCreds(provider, answers) {
+  if (!provider.requiresCreds() ||
+      answers[consts.providerUseExistingKey] ||
+      answers[consts.credsConfirmOverwrite] === false) {
+    return;
+  }
+
+  const credentialsDest = provider.getCredsPath();
+  makeDirPath(provider.getCredsPath());
+
+  if (provider.hasCredsTemplate()) {
+    const templateFile = path.join(templateDir, provider.getCredsTemplate());
+    createFileFromTemplate(templateFile, answers, credentialsDest);
+  } else {
+    // We were given a file path instead of the keys themselves.
+    const credentialsSrc = answers[consts.inputCredsPath];
+    fsExtra.copy(credentialsSrc, credentialsDest, (err) => {
+      if (err) throw err;
+    });
+  }
+
+  util.logExceptOnTest(`Wrote credentials to ${credentialsDest}`);
+}
+
+/**
+  * Retrieve the correct SSH key based on the user's input. This can be no key,
+  * a newly Quilt-generated SSH key, or an existing SSH key at user-given path.
+  *
+  * @param {Promise} answers promise that contains the user's answers.
+  * @return {string} The relevant SSH key.
+  */
+function getSshKey(answers) {
+  let keyPath;
+
+  switch (answers[consts.sshKeyOption]) {
+    case consts.skip:
+      return '';
+
+    case consts.sshGenerateKey:
+      let success = sshGenerateKeyPair();
+      if (!success) {
+        throw new Error('failed to create SSH key. Is ssh-keygen installed?');
```

Yeah looking at this now.  I think it'd be cleaner if `sshGenerateKeyPair` returned nothing, and just threw this exception on error.  You wouldn't have to catch it, could just let it bubble up.